### PR TITLE
Fix for issue #1044

### PR DESCRIPTION
**Pull request by BlackmanCC**
unicode systems always add the byte order mark to the xml, while non-unicode does not
so we will always add the byte order mark if it is not in the xml

### CODE REVIEW

**Comment by larshp** [comment]
merged https://github.com/larshp/abapGit/pull/3166, and pulled upstream changes to your branch, correct errors from abaplint should be shown now 😄

**Comment by BlackmanCC** [comment]
On our non-unicode system abapGit now is usable against repositories from unicode systems.

File: src/objects/zcl_abapgit_objects_files.clas.abap [hunk 1]
```diff
@@ -170,8 +170,10 @@ CLASS ZCL_ABAPGIT_OBJECTS_FILES IMPLEMENTATION.
 
   METHOD add_xml.
 
-    DATA: lv_xml  TYPE string,
-          ls_file TYPE zif_abapgit_definitions=>ty_file.
+    DATA: lv_xml        TYPE string,
+          lv_hex(1)     TYPE x VALUE '23',
```
Comment (larshp) [inline src/objects/zcl_abapgit_objects_files.clas.abap]: suggest

```suggestion
          lv_hex     TYPE x LENGTH 1 VALUE '23',
```

similar for the next definition

Comment (BlackmanCC) [reply src/objects/zcl_abapgit_objects_files.clas.abap]: ok. I will change

**Comment by BlackmanCC** [comment]
@larshp Sorry, it is my second contribution project. Are there any further steps left for me?

**Comment by larshp** [comment]
also change `lv_hex_bom(3) TYPE x VALUE 'EFBBBF',` to use `LENGTH`, but no, code looks good

I typically leave the pull requests for a few days for people to read / review / try out

**Comment by jrodriguez-rc** [comment]
It looks really good but still adds BOM character for abapgit.xml file:
image
image

This could be because the modification adds BOM character to local serialized file instead remove character from the repository files
image

**Comment by BlackmanCC** [comment]
On our system it looks like this:
image

**Comment by jrodriguez-rc** [comment]
It is same I have shown on the first screen, the second screen is a file (package) that doesn't detect changes because of BOM character is added in local serialized file instead remove in remote file

**Comment by BlackmanCC** [comment]
@jrodriguez-rc Please check again.

**Comment by BlackmanCC** [comment]
@larshp maybe it would be better to include the new logic for BOM into zcl_abapgit_convert=>string_to_xstring_utf8( lv_xml ).
What do you mean?

**Comment by larshp** [comment]
zcl_abapgit_convert: yea, but then I suggest adding it as a new method, string_to_xstring_utf8 I would expect to do raw conversion

**Comment by BlackmanCC** [comment]
new method like "fix_bom_for_nonunicode"?

**Comment by larshp** [comment]
perhaps `string_to_xstring_utf8_with_bom` ?

**Pull request merged**